### PR TITLE
Exit early if there's no validator slashing

### PR DESCRIPTION
**Pull request by terencechain**
These validators copies for process slashing are bad. We shouldn't need to copy validators if slashing is not going to happen. This PR implemented a pre condition and a circuit breaker to exit early if there's no meaningful slashing to process

Heap flames for references:
Screen Shot 2020-10-20 at 4 08 00 PM

Related #7544

### CODE REVIEW

File: beacon-chain/core/epoch/epoch_processing.go [hunk 1]
```diff
@@ -248,6 +248,16 @@ func ProcessFinalUpdates(state *stateTrie.BeaconState) (*stateTrie.BeaconState,
 	upwardThreshold := hysteresisInc * params.BeaconConfig().HysteresisUpwardMultiplier
 
 	bals := state.Balances()
+	var updateEffective bool
```
Comment (terencechain) [inline beacon-chain/core/epoch/epoch_processing.go:251]: Note, this should be in a different PR

**Comment by terencechain** [comment]
Related #7585

**Review by rauljordan — APPROVED** [review]
